src/sentry/sentry_apps/services/hook/impl.py

```diff
@@ -59,3 +63,33 @@ def create_service_hook(
                     hook.add_project(project_id)
 
             return serialize_service_hook(hook)
+
+    def replace_service_hook_projects(
+        self, installation_id: int, project_ids: list[int]
+    ) -> list[RpcServiceHookProject]:
+
+        with transaction.atomic(router.db_for_write(ServiceHookProject)):
+            hook = ServiceHook.objects.get(installation_id=installation_id)
+            ServiceHookProject.objects.filter(service_hook_id=hook.id).delete()
+            hook_projects = []
+            for project_id in project_ids:
+                hook_projects.append(
+                    ServiceHookProject.objects.create(
+                        project_id=project_id,
+                        service_hook_id=hook.id,
+                    )
+                )
```

Comment:
This should do a diff instead. So look at the list of expected projects, existing projects, then create/delete based on what has changed.

Reply:
hm not sure if `update_or_create` fits this use case... I'm just going to implement from scratch

Reply:
done